app/javascript/vue/shared/content/checklist.vue

```diff
@@ -117,36 +116,37 @@
     data() {
       return {
         checklistItems: [],
-        linesToPaste: 0,
         editingName: false,
         reordering: false,
-        editingItem: false
+        editingItem: false,
       }
     },
     created() {
-      this.initChecklistItems();
+      this.loadChecklistItems();
 
       if (this.isNew) {
-        this.addItem();
+        this.addItem(orderedChecklistItems.length + 1);
       }
     },
     watch: {
       element() {
-        this.initChecklistItems();
+        this.loadChecklistItems();
       }
     },
     computed: {
       orderedChecklistItems() {
-        return this.checklistItems.map((item, index) => {
-          return { attributes: {...item.attributes, position: index } }
-        });
-      },
-      pastingMultiline() {
-        return this.linesToPaste > 0;
+        return this.checklistItems.sort((a, b) => a.attributes.position - b.attributes.position || b.id - a.id)
```

Comment:
Not sure I understand this sort

Reply:
This for insert new element between other. new element inherit position,  this `id` sorting will verify so element will be after element. This logic to do not move other elements each time.